frontends/mit-learn/src/page-components/SearchField/SearchField.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react"
+import { SearchInput } from "ol-components"
+import type { SearchInputProps, SearchSubmissionEvent } from "ol-components"
+import { usePostHog } from "posthog-js/react"
+
+type SearchFieldProps = SearchInputProps & {
+  onSubmit: (event: SearchSubmissionEvent) => void
+  setPage: (page: number) => void
+}
+
+const { POSTHOG } = APP_SETTINGS
+
+/**
+ * A wrapper around SearchInput that handles a little application logic like
+ * - resetting search page to 1 on submission
+ * - firing tracking events
+ */
+const SearchField: React.FC<SearchFieldProps> = ({
+  onSubmit,
+  setPage,
+  ...others
+}) => {
+  const posthog = usePostHog()
+  const handleSubmit: SearchInputProps["onSubmit"] = (
+    event,
+    { isEnter } = {},
+  ) => {
+    onSubmit(event)
+    setPage(1)
+    if (POSTHOG?.api_key) {
+      posthog.capture("search_update", { isEnter: isEnter })
+    }
```

Comment:
I moved the posthog stuff out of `ol-components` and into the new `SearchField` component. Since ol-components is generally styling / UI focused, I think this new component is a better place for the posthog call.